src/sdk/noobaa_s3_client/noobaa_s3_client.js

```diff
@@ -0,0 +1,148 @@
+/* Copyright (C) 2023 NooBaa */
+'use strict';
+
+const { S3ClientSDKV2 } = require('./noobaa_s3_client_sdkv2');
+const url = require('url');
+const { S3 } = require("@aws-sdk/client-s3");
+const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");
+const dbg = require('../../util/debug_module')(__filename);
+dbg.set_process_name('noobaa_s3_client_by_sdk');
+const config = require('../../../config');
+const http_utils = require('../../util/http_utils');
+
+// The params are the AWS SDK V2 params.
+// params = a map of parameters that are passed to the constructor of S3 Object in order to to bind to every request
+function get_s3_client_v3_params(params) {
+    let s3_client;
+
+    // we use config.AWS_SDK_VERSION_3_DISABLED as workaround in case we want to use only AWS SDK V2
+    if (should_use_sdk_v2(params) || config.AWS_SDK_VERSION_3_DISABLED) {
+        change_s3_client_params_to_v2_structure(params);
+        s3_client = new S3ClientSDKV2(params);
+    } else {
+        s3_client = new S3(params);
+    }
+    return s3_client;
+}
+
+// Here we pack all the functions that would help us decide which version of the sdk should we use
+// How we choose the AWS SDK version for s3 Client?
+// 1. signatureVersion defined? if it is 'v2' => AWS SDK V2
+// 2. maxRedirects defined? if it is a non-negative number => AWS SDK V2
+// 3. AWS SDK V3
+function should_use_sdk_v2(params) {
+    if (params.signatureVersion) {
+        const use_sdk_v2_by_signature_version = should_use_sdk_v2_by_signature_version(params.signatureVersion);
+        if (use_sdk_v2_by_signature_version) return use_sdk_v2_by_signature_version;
+    }
+
+    if (params.maxRedirects) {
+        const use_sdk_v2_by_max_redirects = should_use_sdk_v2_by_max_redirects(params.maxRedirects);
+        return use_sdk_v2_by_max_redirects;
+    }
+
+    return false;
+}
```

Comment:
This function should be more straightforward. 
First, I don't think we should take `maxRedirects` as a factor. We use it only in 2 tests, and even then, we explicitly set it to 10, which is the default value. 
Also, there is no need to break each condition into its own function. It just makes the code very verbose. 

```suggestion
function should_use_sdk_v2(params) {
    return config.AWS_SDK_VERSION_3_DISABLED || params?.signatureVersion
}
```

Reply:
I missed the `=== 'v2'`

Regarding the condition, in general, in an `or` statement, if the first conditional is true, the second one will not be evaluated. For an `and` statement, if the first is `false` the second is never evaluated.